examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp

```diff
@@ -40,24 +40,28 @@
 #include <logging/log.h>
 #include <zephyr.h>
 
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
```

Comment:
I think we shouldn't use `CONFIG_MATTER_LOG_LEVEL` to control log level of the application. It is for controlling Matter library.

Reply:
Hmm, so what should we use instead of that? As I remember it was added as a workaround because we used LOG notation from Zephyr in our samples and we wanted to have the same log level in application and Matter library.

Reply:
It's of course fine to use LOG_ macros for the application but what is not right is using Matter's log level to configure the application's log level. It's like BLE samples using log level of one of BLE host or BLE controller to configure their log level. It's just unnecessary coupling. We can pass harcoded `LOG_LEVEL_DBG` (like some other samples in sdk-nrf do) or we can define another Kconfig in each sample to configure the log level. I once needed to reduce the flash size of Matter samples so I disabled Matter logs, and was very surprised that the application logs disappeared as well :).

Reply:
Ok, I see the point :) I've not thought about it, seen in NCS samples `LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);`, and just copied this line. I agree with you completely, and will add Kconfig to control the application log level.